recipe-client-addon/lib/PreferenceManagement.jsm

```diff
@@ -0,0 +1,395 @@
+ */
+"use strict";
+
+const {utils: Cu} = Components;
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "Services", "resource://gre/modules/Services.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "CleanupManager", "resource://shield-recipe-client/lib/CleanupManager.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "JSONFile", "resource://gre/modules/JSONFile.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "OS", "resource://gre/modules/osfile.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "LogManager", "resource://shield-recipe-client/lib/LogManager.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Preferences", "resource://gre/modules/Preferences.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "TelemetryEnvironment", "resource://gre/modules/TelemetryEnvironment.jsm");
+
+this.EXPORTED_SYMBOLS = ["PreferenceManagement"];
+
+const STORE_FILE = "shield-preference-recipes.json";
+
+const PREFERENCE_TYPE_MAP = {
+  boolean: Services.prefs.PREF_BOOL,
+  string: Services.prefs.PREF_STRING,
+  integer: Services.prefs.PREF_INT,
+};
+
+const DefaultPreferences = new Preferences({defaultBranch: true});
+/**
+ * Enum storing Preference modules for each type of preference branch.
+ * @enum {Object}
+ */
+const PreferenceBranchType = {
+  user: Preferences,
+  default: DefaultPreferences,
+};
+
+const log = LogManager.getLogger("preference-management");
+
+this.PreferenceManagement = (namespace) => ({
+  preferenceObservers: new Map(),
+
+  // Internal 'store loaded' tracking object
+  storePromise: undefined,
+
+  /**
+   * Asynchronously loads the local JSON file used for storage, returning the
+   * request Promise.
+   *
+   * @returns {Promise}
+   * @resolves When the operation is completed succesfully
+   * @rejects Javascript exception.
+   */
+  ensureStorage() {
+    if (this.storePromise === undefined) {
+      const path = OS.Path.join(OS.Constants.Path.profileDir, STORE_FILE);
+      const storage = new JSONFile({path});
+      this.storePromise = storage.load().then(() => storage);
+    }
+    return this.storePromise;
+  },
+
+  /**
+   * Returns the internal storage object, after properly ensuring the store and
+   * store namespace are initiated.
+   *
+   * @returns {Promise}
+   * @resolves When the operation is completed succesfully
+   * @rejects Javascript exception.
+   */
+  async getStorage() {
+    const store = await this.ensureStorage();
+    store.data = store.data || {};
+    store.data[namespace] = store.data[namespace] || {};
+    store.saveSoon();
+
+    return store;
+  },
+
+  /**
+   * Clears the namespaced storage.
+   *
+   * @returns {Promise}
+   * @resolves When the operation is completed succesfully
+   * @rejects Javascript exception.
+   */
+  async clearStorage() {
+    const store = await this.getStorage();
+    store.data[namespace] = {};
+    store.saveSoon();
+  },
+
+
+  /**
+   * Returns the internal storage object, after properly ensuring the store and
```

Comment:
You're right - this function was actually phased out during dev and should have been removed. Sorry for the noise.